### PR TITLE
Fix bug pipeline details page for Pipeline with no tasks throws error

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix bug - https://jira.coreos.com/browse/ODC-1858

Peek 2019-09-16 12-11

### CODE REVIEW

**Comment by andrewballantyne** [comment]
@vikram-raj I think we probably want to drop that grey circle under the Overview header.

I think we probably want to explain why it doesn't have an overview, so maybe something like:

image

Will need UX for the verbiage (cc @serenamarie125 )

I am leaning towards thinking we should keep the title so there isn't just a void in "why don't I see the visualization on _this_ pipeline?" kind of questions. Thoughts Serena?

**Comment by vikram-raj** [comment]
@andrewballantyne Agree, and there is a separate [issue](https://jira.coreos.com/browse/ODC-1860) for that and we need confirmation from the issue. Once UX will confirm it then I will remove that.

File: frontend/packages/dev-console/src/components/pipelines/PipelineDetails.tsx [hunk 1]
```diff
@@ -12,28 +12,30 @@ const PipelineDetails = ({ obj: pipeline }) => (
...
-                    title={task.taskRef.name}
-                    inline
-                  />
-                </dd>
-              </React.Fragment>
-            );
-          })}
-        </dl>
-      </div>
+      {pipeline.spec.tasks && (
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/pipelines/PipelineDetails.tsx]: `pipeline.spec` nor `pipeline.spec.tasks` are guaranteed by the type. Recommend handling the NPEs as well.

Also, could we get an explicit type on this variable - like type the FC?

Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/pipelines/PipelineDetails.tsx]: In `frontend/public/module/k8s/index.ts`, this type is used for other components that this component calls.
```
export type K8sResourceKind = {
  apiVersion?: string;
  kind?: string;
  metadata?: ObjectMetadata;
  spec?: {
    selector?: Selector;
    [key: string]: any
  };
  status?: {[key: string]: any};
  type?: {[key: string]: any};
  data?: {[key: string]: any};
};
```

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/pipelines/PipelineDetails.tsx]: @andrewballantyne Done.

**Comment by andrewballantyne** [comment]
Well, since the design stuff is tracked in another ticket - I'll just review the problem at hand... and that was a NPE breaking things. You fixed that, so I guess that's pretty much it for this PR.

/lgtm
/test e2e-aws-console

**Comment by vikram-raj** [comment]
@andrewballantyne Yeah, we get the input from the UX then I will fix the design issue as well.

**Pull request merged**